### PR TITLE
Hide tab bars

### PR DESCRIPTION
**Pull request by mvbattan**
#### Default behavior
tabbardefault

#### No-hide behavior (passing `alwaysShowTabBar`)
tabbarnohide

### CODE REVIEW

File: src/TabsContainer.js [hunk 1]
```diff
@@ -21,6 +22,16 @@ class TabsContainer extends Component {
         {...this.props.tabsComponentProps}
         onChangeTab={this.handleTabChanged}
         initialPage={this.initialTab}
+        renderTabBar={(props) => {
+          if (this.props.shouldHideTabBar) {
+            return <View />;
```
Comment (mvbattan) [inline src/TabsContainer.js]: Doesn't work since `null` has not props to get.

File: src/RootScene.js [hunk 2]
```diff
@@ -72,6 +74,17 @@ export default class RootScene extends Component {
     return route.transition || this.props.defaultTransition;
   }
 
+  handleRouteChange = () => {
+    const navigator = this.getNavigator();
+    // Stack diff can be either 0 or 2, but never 1.
+    if (
+      (navigator && this.routeStack && navigator.state.routeStack.length) ===
```
Comment (sbalay) [inline src/RootScene.js]: why these parenthesis?
we should check that the navigator exists, then check if we have tracked the routeStack. If those are ok, then check if the stacks length are ok.
This should be
`if (navigator && this.routeStack && navigator.state.routeStack.length === this.routeStack.length + 1)`

File: src/RootScene.js [hunk 3]
```diff
@@ -72,6 +74,17 @@ export default class RootScene extends Component {
...
+      (this.routeStack.length + 1)
+    ) {
+      this.props.dispatch(actionCreators.pop());
+    }
```
Comment (sbalay) [inline src/RootScene.js]: I'd set `this.routeStack` to null here to make it clear that that variable is only meant to be used here.

Comment (sbalay) [inline src/RootScene.js]: In any other place we should get the routeStack from props

File: src/TabsContainer.js [hunk 4]
```diff
@@ -21,6 +22,16 @@ class TabsContainer extends Component {
         {...this.props.tabsComponentProps}
         onChangeTab={this.handleTabChanged}
         initialPage={this.initialTab}
+        renderTabBar={(props) => {
+          if (this.props.shouldHideTabBar && !this.props.alwaysShowTabBar) {
+            return <View />;
+          }
+          if (this.props.tabsComponentProps.renderTabBar) {
+            return this.props.tabsComponentProps.renderTabBar();
```
Comment (sbalay) [inline src/TabsContainer.js]: this function should also receive `props` as param.

**Pull request merged**